content/docs/glossary/transactions.mdx

```diff
@@ -42,10 +42,39 @@ The memo contains optional extra information. It is the responsibility of the cl
 - `MEMO_HASH` : A 32 byte hash.
 - `MEMO_RETURN` : A 32 byte hash intended to be interpreted as the hash of the transaction the sender is refunding.
 
-### Time Bounds
+### Validity Conditions
 
+There are a number of ways to control when a transaction should be considered valid. These conditions are checked first, prior to the validity of the operations or other parts of the transaction, so they're often referred to as "preconditions". They can all be combined, provided the combination is logically sound.
+
+For some of these, it's important to understand a related concept: the last time an account consumed a sequence number. Both the ledger number and timestamp of the last time this occurred is tracked by the network, and these values enable some of the preconditions below.
+
+#### Time Bounds
 The optional UNIX timestamp (in seconds), determined by ledger time, of a lower and upper bound of when this transaction will be valid. If a transaction is submitted too early or too late, it will fail to make it into the transaction set. `maxTime` equal `0` means that it's not set. _We highly advise for all transactions to use time bounds, and many SDKs enforce their usage._ If a transaction doesn't make it into the transaction set, it is kept around in memory in order to be added to the next transaction set on a best-effort basis. Because of this behavior, we highly advise that all transactions are created with time bounds in order to invalidate transactions after a certain amount of time, especially if you plan to resubmit your transaction at a later time.
 
+#### Ledger Bounds
+These are like [Time Bounds](#time-bounds), except they apply to ledger numbers. With them set, a transaction will only be valid for ledger numbers that fall into the range you set. The lower bound is inclusive while the upper bound is not. If you set the upper bound to zero, this indicates that there is no upper bound.
```

Comment:
>The lower bound is inclusive while the upper bound is not.

This is incorrect. The upper bound is inclusive.

See https://github.com/stellar/stellar-core/blob/da846c923bbd989fe716c9a3eb32e04ef2ae37db/src/transactions/TransactionFrame.cpp#L491-L496.

cc @sisuresh for confirmation

Reply:
Then the [CAP-21 comments](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0021.md#xdr-diff) are incorrect:

```diff
+    // Transaction only valid for ledger numbers n such that
+    // minLedger <= n < maxLedger (if maxLedger == 0, then
+    // only minLedger is checked)
+    LedgerBounds *ledgerBounds;
```